src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,138 @@
+                "Registered a user with age less than minimum without throwing an exception");
+    }
+
+    @Test
+    void register_NegativeAge_NotOk() {
+        correctUser.setAge(-1);
+        assertThrows(RegistrationException.class, () -> regService.register(correctUser),
+                "Registered a user with negative age value without throwing an exception");
+    }
+
+    @Test
+    void register_ZeroAge_NotOk() {
+        correctUser.setAge(0);
+        assertThrows(RegistrationException.class, () -> regService.register(correctUser),
+                "Registered a user with zero age without throwing an exception");
+    }
+
+    @Test
+    void register_NullAge_NotOk() {
+        correctUser.setAge(null);
+        assertThrows(RegistrationException.class, () -> regService.register(correctUser),
+                "Registered a user with age null without throwing an exception");
+    }
+
+    @Test
+    void register_AgeMin_Ok() {
+        correctUser.setAge(MIN_AGE);
+        assertEquals(correctUser, regService.register(correctUser),
+                "User with minimum age wasn't registered");
+        assertEquals(correctUser, storageDao.get(correctUser.getLogin()),
+                "User with minimum age wasn't added to database");
+    }
+
+    @Test
+    void register_AgeOverMin_Ok() {
+        correctUser.setAge(MIN_AGE + 1);
+        assertEquals(correctUser, regService.register(correctUser),
+                "User with age above minimum wasn't registered");
+        assertEquals(correctUser, storageDao.get(correctUser.getLogin()),
+                "User with age above minimum wasn't added to database");
+    }
+
+    @Test
+    void register_NullLogin_NotOk() {
+        correctUser.setLogin(null);
+        assertThrows(RegistrationException.class, () -> regService.register(correctUser),
+                "Registered a user with Login null without throwing an exception");
+    }
+
+    @Test
+    void register_ZeroLengthLogin_NotOk() {
+        correctUser.setLogin("");
+        assertThrows(RegistrationException.class, () -> regService.register(correctUser),
+                "Registered a user with empty Login without throwing an exception");
+    }
+
+    @Test
+    void register_NullPassword_NotOk() {
+        correctUser.setPassword(null);
+        assertThrows(RegistrationException.class, () -> regService.register(correctUser),
+                "Registered a user with null password without throwing an exception");
+    }
+
+    @Test
+    void register_ShortPassword_NotOk() {
+        correctUser.setPassword(TOO_SHORT_PASSWORD);
+        assertThrows(RegistrationException.class, () -> regService.register(correctUser),
+                "Registered a user with password shorter than minimum length "
+                        + "without throwing an exception");
+    }
+
+    @Test
+    void register_UserAlreadyAdded_NotOk() {
+        storageDao.add(correctUser);

```

Comment:
we should use as low-level methods as possible
```suggestion
        Storage.people.add(correctUser);
```